be/src/formats/parquet/column_chunk_reader.cpp

```diff
@@ -51,19 +52,56 @@ Status ColumnChunkReader::init(int chunk_size) {
     RETURN_IF_ERROR(get_block_compression_codec(compress_type, &_compress_codec));
 
     RETURN_IF_ERROR(_try_load_dictionary(chunk_size));
-    RETURN_IF_ERROR(_parse_page_data());
     return Status::OK();
 }
 
 Status ColumnChunkReader::next_page() {
     RETURN_IF_ERROR(_parse_page_header());
-    RETURN_IF_ERROR(_parse_page_data());
+    return Status::OK();
+}
+
+Status ColumnChunkReader::load_page() {
+    if (_page_parse_state == PAGE_DATA_PARSED) {
+        return Status::OK();
+    }
+    if (_page_parse_state != PAGE_HEADER_PARSED) {
+        return Status::InternalError("Page header has not been parsed before loading page data");
+    }
+    return _parse_page_data();
+}
+
+Status ColumnChunkReader::skip_page() {
+    if (_page_parse_state == PAGE_DATA_PARSED) {
+        return Status::OK();
+    }
+    if (_page_parse_state != PAGE_HEADER_PARSED) {
+        return Status::InternalError("Page header has not been parsed before skiping page data");
+    }
+    const auto& header = *_page_reader->current_header();
+    uint32_t compressed_size = header.compressed_page_size;
+    uint32_t uncompressed_size = header.uncompressed_page_size;
+    size_t size = _compress_codec != nullptr ? compressed_size : uncompressed_size;
+    RETURN_IF_ERROR(_page_reader->skip_bytes(size));
+
+    _page_parse_state = PAGE_DATA_PARSED;
     return Status::OK();
 }
 
 Status ColumnChunkReader::_parse_page_header() {
```

Comment:
Yes, rename it to `load_page_header`.

Reply:
> Yes, rename it to `load_page_header`.

The word 'parse' can correspond to the state 'PAGE_HEADER_PARSED'.